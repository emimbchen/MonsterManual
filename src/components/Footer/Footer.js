import { ReactComponent as LinkedInIcon } from "../../assets/linkedin.svg";
import { ReactComponent as GithubIcon } from "../../assets/github.svg";

function Footer({ ...props }) {
  return (
    <footer role="contentinfo" className="bg-primary p-6" {...props}>
      <div className="flex justify-center gap-4 py-5">
        <a href="https://github.com/emimbchen">
          <LinkedInIcon className="h-9 w-auto" />
          <span className="sr-only">Emi's Linkedin</span>
        </a>
        <a href="https://www.linkedin.com/in/emi-mb-chen/">
          <GithubIcon className="h-9 w-auto" />
          <span className="sr-only">Emi's Github</span>
        </a>
      </div>
      <ul className="text-white">
        <li>
          Favicon from:
          <a
            href="https://thenounproject.com/browse/icons/term/card/"
            target="_blank"
            title="Card Icons"
            className="underline ms-2"
          >
            Noun Project
          </a>
        </li>
        <li>
          January of hwatu icon by 1516 from:
          <a
            href="https://thenounproject.com/browse/icons/term/january-of-hwatu/"
            target="_blank"
            title="January of hwatu Icons"
            className="underline ms-2"
          >
            Noun Project
          </a>
        </li>
      </ul>
    </footer>
  );
}

export default Footer;
