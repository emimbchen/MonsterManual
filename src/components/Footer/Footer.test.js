import axe from "axe-core";
import Footer from "./Footer";
import { render } from "@testing-library/react";

describe("Footer Component", () => {
  test("passes accessibility check", async () => {
    const { container } = render(<Footer />);
    const results = await axe.run(container);
    expect(results.violations.length).toBe(0);
  });

  it("renders correctly", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
