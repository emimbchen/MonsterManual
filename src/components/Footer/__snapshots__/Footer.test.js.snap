// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders correctly 1`] = `
<DocumentFragment>
  <footer
    class="bg-primary p-6"
    role="contentinfo"
  >
    <div
      class="flex justify-center gap-4 py-5"
    >
      <a
        href="https://github.com/emimbchen"
      >
        <svg
          class="h-9 w-auto"
        >
          linkedin.svg
        </svg>
        <span
          class="sr-only"
        >
          Emi's Linkedin
        </span>
      </a>
      <a
        href="https://www.linkedin.com/in/emi-mb-chen/"
      >
        <svg
          class="h-9 w-auto"
        >
          github.svg
        </svg>
        <span
          class="sr-only"
        >
          Emi's Github
        </span>
      </a>
    </div>
    <ul
      class="text-white"
    >
      <li>
        Favicon from:
        <a
          class="underline ms-2"
          href="https://thenounproject.com/browse/icons/term/card/"
          rel="noreferrer"
          target="_blank"
          title="Card Icons"
        >
          Noun Project
        </a>
      </li>
      <li>
        January of hwatu icon by 1516 from:
        <a
          class="underline ms-2"
          href="https://thenounproject.com/browse/icons/term/january-of-hwatu/"
          rel="noreferrer"
          target="_blank"
          title="January of hwatu Icons"
        >
          Noun Project
        </a>
      </li>
    </ul>
  </footer>
</DocumentFragment>
`;
